toolz/functoolz.py

```diff
@@ -174,8 +175,12 @@ def __init__(self, func, *args, **kwargs):
         else:
             self._partial = partial(func, *args)
 
-        self.__doc__ = getattr(func, '__doc__', None)
-        self.__name__ = getattr(func, '__name__', '<curry>')
+        self.__name__ = getattr(func, '__name__', '<curried>')
+        self.__doc__ = getattr(func, '__doc__', '<curry>')
+        self.__module__ = getattr(func, '__module__', '')
```

Comment:
Would it make sense to default to `self.__module__`, which would result in "toolz.functoolz"?